src/eduid_idp/idp_actions.py

```diff
@@ -94,24 +96,31 @@ def check_for_pending_actions(idp_app, user, ticket, sso_session):
     idp_app.logger.debug('There are pending actions for user {}: {}'.format(user, pending_actions))
 
     # create auth token for actions app
-    secret = idp_app.config.actions_auth_shared_secret
-    nonce = os.urandom(16)
+    eppn = user.eppn
+    shared_key = idp_app.config.actions_auth_shared_secret
+    if not isinstance(shared_key, six.binary_type):
+        shared_key = shared_key.encode('ascii')
+    timestamp = '{:x}'.format(int(time.time()))
+    nonce = nacl.utils.random(nacl.secret.SecretBox.NONCE_SIZE)
+    token_data = '{0}|{1}'.format(timestamp, eppn).encode('ascii')
+    box = nacl.secret.SecretBox(shared_key)
+    encrypted = box.encrypt(token_data, nonce)
```

Comment:
Better not pass a nonce to box.encrypt. Doing so just enables foot-shooting. It will generate a random nonce if none is provided.

Reply:
fixed